README.md

```diff
@@ -3113,13 +3222,124 @@ MIDDLEWARE = [
+
+It is actually `render_dependencies` that finds all used components in the HTML string,
+and inserts the component's JS and CSS into `{% component_dependencies %}` tags, or at the default locations.
+
+#### Render JS / CSS without the middleware
+
+The `ComponentDependencyMiddleware` middleware just calls `render_dependencies`, passing in the HTML
+content. So if you rendered a template that contained `{% components %}` tags, instead of the middleware,
+you MUST pass the result through `render_dependencies`:
+
+```py
+from django.template.base import Template
+from django.template.context import Context
+from django_component import render_dependencies
+
+template = Template("""
+    {% load component_tags %}
+    <!doctype html>
+    <html>
+    <head>
+        <title>MyPage</title>
+    </head>
+    <body>
+        <main>
+            {% component "my_button" %}
+                Click me!
+            {% endcomponent %}
+        </main>
+    </body>
+    </html>
+""")
+
+rendered = template.render(Context({}))
+rendered = render_dependencies(rendered)
+```
+
+Same applies if you render a template using Django's [`django.shortcuts.render`](https://docs.djangoproject.com/en/5.1/topics/http/shortcuts/#render):
+
+```py
+from django.shortcuts import render
+
+def my_view(request):
+    rendered = render(request, "pages/home.html")
+    rendered = render_dependencies(rendered)
+    return rendered
+```
+
+Alternatively, when you render HTML with `Component.render()` or `Component.render_to_response()`,
+these automatically call `render_dependencies()` for you, so you don't have to:
+
+```py
+from django_components import Component
+
+class MyButton(Component):
+    ...
+
+# No need to call `render_dependencies()`
+rendered = MyButton.render()
+```
+
+#### Inserting pre-rendered HTML into another component
+
+In previous section we've shown that `render_dependencies()` does NOT need to be called
+when you render a component via `Component.render()`.
+
+API of django-components makes it possible to compose components in a "React-like" way,
+where we pre-render a piece of HTML and then insert it into a larger structure:
+
+```py
+card_actions = CardActions.render(
+    kwargs={"editable": editable},
+)
+
+card = Card.render(
+    slots={"actions": card_actions},
+)
+
+page = MyPage.render(
+    slots={"card": card},
+)
+```
+
+In this case, as mentioned above, each time we call `Component.render()`, we also call `render_dependencies()`.
+
+However, there is a problem here - When we call `render_dependencies()` inside `CardActions.render`,
+we extract the info on components' JS and CSS from the HTML. But the template of `CardActions`
+contains no `{% component_depedencies %}` tags, and nor `<head>` nor `<body>` HTML tags.
+So the component's JS and CSS will NOT be inserted, and will be lost.
+
+To work around this, you must set `kind=None` when rendering pieces of HTML with `Component.render()`
```

Comment:
So I tried different approaches today, but TBH all of them are unnecessarily complex and felt error prone, or would require that we re-parse the HTML every time we insert one component into another.

So that means that we DO need to pass some kind of flag to `Component.render()` to tell it that the component will be nested into another one.

So what about `nested=True`?

Reply:
What about render_dependencies=False?